src/main/java/com/google/cloud/healthcare/etl/runner/dicomtofhir/DicomToFhirStreamingRunner.java

```diff
@@ -0,0 +1,133 @@
+package com.google.cloud.healthcare.etl.runner.dicomtofhir;
+
+import com.google.cloud.healthcare.etl.model.ErrorEntry;
+import com.google.cloud.healthcare.etl.model.mapping.HclsApiDicomMappableMessage;
+import com.google.cloud.healthcare.etl.model.mapping.HclsApiDicomMappableMessageCoder;
+import com.google.cloud.healthcare.etl.pipeline.MappingFn;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.io.gcp.healthcare.DicomIO;
+import org.apache.beam.sdk.io.gcp.healthcare.FhirIO;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubIO;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.options.Description;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.options.Validation.Required;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionTuple;
+import org.apache.beam.sdk.values.TupleTagList;
+import org.apache.beam.sdk.values.TypeDescriptor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+
+public class DicomToFhirStreamingRunner {
+    private static final Logger LOG = LoggerFactory.getLogger(DicomToFhirStreamingRunner.class);
+
+    public static class ExtractWebpathFromPubsub extends DoFn<PubsubMessage, String> {
```

Comment:
You can add tests for these phases: sample test: https://github.com/GoogleCloudPlatform/healthcare-data-harmonization-dataflow/blob/master/src/test/java/com/google/cloud/healthcare/etl/pipeline/MappingFnTest.java